modules/01-mods.js

```diff
@@ -1,538 +1,434 @@
+  const reason = interaction.options.getString("reason", true);
+  const value = interaction.options.getInteger("value") ?? 1;
+  if (!member) return interaction.editReply(noTarget);
+
+  const warn = await c.warn(interaction, reason, value, member);
+  return interaction.editReply(warn);
 }
 
+/** @param {Augur.GuildInteraction<"CommandSlash">} interaction*/
+async function slashModGrownups(interaction) {
+  const time = Math.min(30, interaction.options.getInteger("time") ?? 15);
+  if (!interaction.channel) return interaction.reply({ content: "Well that's awkward, I can't access the channel you're in!", ephemeral: true });
+  if ((interaction.channel.parentId || "") != u.sf.channels.staffCategory) return interaction.reply({ content: "This command can only be used in the LDSG-Staff Category!", ephemeral: true });
+  interaction.reply(time == 0 ? `*Whistles and wanders back in*` : `*Whistles and wanders off for ${time} minutes...*`);
+
+  if (c.grownups.has(interaction.channel.id)) {
+    clearTimeout(c.grownups.get(interaction.channel.id));
+  }
+  if (time == 0) {
+    c.grownups.delete(interaction.channel.id);
+  } else {
+    c.grownups.set(interaction.channel.id, setTimeout((channel) => {
+      c.grownups.delete(channel.id);
+      channel.send("*I'm watching you again...* :eyes:");
+    }, time * 60 * 1000, interaction.channel));
+  }
+}
+
+
 Module.addInteraction({
   name: "mod",
-  guild: u.sf.ldsg,
+  guildId: u.sf.ldsg,
   id: u.sf.commands.slashMod,
-  permissions: p.isMod,
-  /** @param {Discord.CommandInteraction} interaction*/
-  process: async (interaction) => {
+  onlyGuild: true,
+  permissions: (int) => p.calc(int.member, ["mod"]),
+  process: (interaction) => {
     try {
       const subcommand = interaction.options.getSubcommand(true);
       switch (subcommand) {
-      case "ban":
-        await slashModBan(interaction);
-        break;
-      case "filter":
-        await slashModFilter(interaction);
-        break;
-      case "fullinfo":
-        await slashModFullInfo(interaction);
-        break;
-      case "kick":
-        await slashModKick(interaction);
-        break;
-      case "mute":
-        await slashModMute(interaction);
-        break;
-      case "note":
-        await slashModNote(interaction);
-        break;
-      case "office":
-        await slashModOffice(interaction);
-        break;
-      case "purge":
-        await slashModPurge(interaction);
-        break;
-      case "rename":
-        await slashModRename(interaction);
-        break;
-      case "slowmode":
-        await slashModSlowmode(interaction);
-        break;
-      case "summary":
-        await slashModSummary(interaction);
-        break;
-      case "trust":
-        await slashModTrust(interaction);
-        break;
-      case "warn":
-        await slashModWarn(interaction);
-        break;
+      case "ban": return slashModBan(interaction);
```

Comment:
Argument of type 'ChatInputCommandInteraction<CacheType>' is not assignable to parameter of type 'ChatInputCommandInteraction<"cached">'.
  Type 'CacheType' is not assignable to type '"cached"'.
    Type 'undefined' is not assignable to type '"cached"'.ts(2345)
    
    on 'interaction' argument